src/ui-test/tests/custom_settings.test.ts

```diff
@@ -0,0 +1,103 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { assert } from 'chai';
+import { SettingsEditor, Workbench, VSBrowser, TextSetting, TextEditor } from 'vscode-uitests-tooling';
+import * as path from 'path';
+import * as utils from '../utils/testUtils';
+
+describe('Test custom settings', function () {
+    this.timeout(60000);
+
+    let settings: SettingsEditor;
+	
+    let contentAssist;
+
+    const RESOURCES = path.resolve('src', 'ui-test', 'resources');
+    const CAMEL_CONTEXT_XML = 'camel-context.xml';
+    const URI_POSITION = 33;
+
+    before(async function () {
+        this.timeout(20000);
+        VSBrowser.instance.waitForWorkbench();
+    });
+
+    // reset version to default
+    async function _setDefaultCamelCatalogVersion() {
+        await setCamelCatalogVersion(' ');
+    }
+
+    describe('Camel catalog', function () {
+
+        after(_setDefaultCamelCatalogVersion);
+
+        it('Default version', async function () {
+            // open file
+			await VSBrowser.instance.openResources(path.join(RESOURCES, CAMEL_CONTEXT_XML));
+            
+            // add component
+            const editor = new TextEditor();
+            await editor.isDisplayed();
+            await editor.typeTextAt(3, URI_POSITION, 'file-watch');
+
+            // open ca
+            contentAssist = await new TextEditor().toggleContentAssist(true);
+            const items = await contentAssist.getItems();
+            // should be 1
+            assert.equal(items.length, 1);
+
+            // check if content is expected
+            const expectedContentAssist = 'file-watch:path';
+			const timer = await contentAssist.getItem(expectedContentAssist);
+			assert.equal(await utils.getTextExt(timer), expectedContentAssist);
+
+            // close file w\o saving
+            await utils.closeEditor(CAMEL_CONTEXT_XML, false);
+        });
+
+        it('2.15.1 version', async function () {
+            // set version
+            await setCamelCatalogVersion('2.15.1');
+
+            // open file
+            await VSBrowser.instance.openResources(path.join(RESOURCES, CAMEL_CONTEXT_XML));
+
+            // add component
+            const editor = new TextEditor();
+            await editor.isDisplayed();
+            await editor.typeTextAt(3, URI_POSITION, 'file-watch');
+            
+            // open ca
+            contentAssist = await new TextEditor().toggleContentAssist(true);
+            const items = await contentAssist.getItems();
+            // should be empty
+            assert.equal(items.length, 0);
+
+            // close file w\o saving
+            await utils.closeEditor(CAMEL_CONTEXT_XML, false);
+        }); 
+
+    });
+
+    async function setCamelCatalogVersion(version: string): Promise<void> {
+        settings = await new Workbench().openSettings();
+        const setting = await settings.findSetting('Camel catalog version', 'Camel');
```

Comment:
lines 98 and 99 could be one line

`const textField = await settings.findSetting('Camel catalog version', 'Camel') as TextSetting;`